src/content/blog/2024/nf-test-in-nf-core.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Leveraging nf-test for enhanced quality control in nf-core
+date: 2024-03-28
+type: post
+description: Reproducibility is an important attribute of all good science. This is specially true in the realm of bioinformatics, where software is hopefully being constantly updated, and pipelines are ideally being maintained. This blog post covers nf-test in the nf-core context.
+image: img/blog-2024-03-28--share.jpg
+tags: nextflow,nf-core,nf-test
+status: published
+author: Carson Miller & Sateesh Peri
+icon: carson_sateesh.png
+---
+
+# The ever-changing landscape of bioinformatics
+
+Reproducibility is an important attribute of all good science. This is especially true in the realm of bioinformatics, where software is hopefully being constantly updated, and pipelines are ideally being maintained. Improvements and maintenance are great, but they also bring about an important question: Do bioinformatics tools and pipelines continue to run successfully and produce consistent results despite these changes? Fortunately for us, there is an existing approach to ensure software reproducibility: testing.
+
+<!-- end-archive-description -->
+
+# The Wonderful World of Testing
+
+> "Software testing is the process of evaluating and verifying that a software product does what it is supposed to do,"
+> Lukas Forer, co-creator of nf-test.
+
+Software testing has two primary purposes: determining whether an operation continues to run successfully after changes are made, and comparing outputs to see if they are consistent. Testing can alert the developer that an output has changed so that an appropriate fix can be made. Admittedly, there are some instances when altered outputs are intentional (i.e., improving a tool might lead to better, and therefore different, results). However, even in these scenarios, it is important to know what has changed, so that no unintentional changes are being made as a result of the update.
```

Comment:
```suggestion
Software testing has two primary purposes: determining whether an operation continues to run successfully after changes are made, and comparing outputs across runs to see if they are consistent. Testing can alert the developer that an output has changed so that an appropriate fix can be made. Admittedly, there are some instances when altered outputs are intentional (i.e., improving a tool might lead to better, and therefore different, results). However, even in these scenarios, it is important to know what has changed, so that no unintentional changes are introduced during an update.
```